chasm/lib/scheduler/scheduler_idle_tasks_test.go

```diff
@@ -0,0 +1,156 @@
+package scheduler_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"go.temporal.io/server/chasm"
+	"go.temporal.io/server/chasm/lib/scheduler"
+	"go.temporal.io/server/chasm/lib/scheduler/gen/schedulerpb/v1"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type idleTasksSuite struct {
+	schedulerSuite
+	executor *scheduler.SchedulerIdleTaskExecutor
+}
+
+func TestIdleTasksSuite(t *testing.T) {
+	suite.Run(t, &idleTasksSuite{})
+}
+
+func (s *idleTasksSuite) SetupTest() {
+	s.SetupSuite()
+	// Note: executor will be created per test case with custom config
+}
+
+type idleValidateTestCase struct {
+	ConfigIdleTime           time.Duration
+	TaskIdleTimeTotal        time.Duration
+	ScheduledTime            time.Time
+	SchedulerClosed          bool
+	IdleMatchesScheduledTime bool
+	SetupScheduler           func(*scheduler.Scheduler, chasm.Context)
+	ExpectedValid            bool
+}
+
+func (s *idleTasksSuite) TestExecute() {
+	ctx := s.newMutableContext()
+	sched := s.scheduler
+
+	// Create executor with default config
+	executor := scheduler.NewSchedulerIdleTaskExecutor(scheduler.SchedulerIdleTaskExecutorOptions{
+		Config: defaultConfig(),
+	})
+
+	// Verify scheduler starts open
+	s.False(sched.Closed)
+
+	// Execute the idle task
+	err := executor.Execute(ctx, sched, chasm.TaskAttributes{}, &schedulerpb.SchedulerIdleTask{})
+	s.NoError(err)
+
+	// Verify scheduler is now closed
+	s.True(sched.Closed)
+}
+
+func (s *idleTasksSuite) TestValidate_IdleTimeDecreased() {
+	now := s.timeSource.Now()
+	s.runValidateTestCase(&idleValidateTestCase{
+		ConfigIdleTime:    5 * time.Minute,
+		TaskIdleTimeTotal: 10 * time.Minute,
+		ScheduledTime:     now,
+		ExpectedValid:     true,
+	})
+}
+
+func (s *idleTasksSuite) TestValidate_SchedulerNotIdle() {
+	now := s.timeSource.Now()
+	s.runValidateTestCase(&idleValidateTestCase{
+		ConfigIdleTime:    10 * time.Minute,
+		TaskIdleTimeTotal: 10 * time.Minute,
+		ScheduledTime:     now,
+		SetupScheduler: func(sched *scheduler.Scheduler, ctx chasm.Context) {
+			// Make scheduler not idle by setting it as paused
+			sched.Schedule.State.Paused = true
+		},
+		ExpectedValid: false,
+	})
+}
+
+func (s *idleTasksSuite) TestValidate_IdleExpirationChanged() {
```

Comment:
Seems redundant to test this now that we know for sure that we don't look at the config value but 🤷